File: frontend/shared/src/types/measurements.ts
export interface FullMeasurements {
  back_waist_length: number;
  front_waist_length: number;
  full_bust: number;
  bust_height: number;
  half_bust_point_distance: number;
  full_waist: number;
  small_hip: number;
  full_hip: number;
  neck_circumference: number;
  half_back_width: number;
  half_front_width: number;
  shoulder_length: number;
  armhole_circumference: number;
  underarm_height: number;
  arm_length: number;
  upper_arm: number;
  elbow_height: number;
  wrist: number;
  waist_to_hip: number;
  crotch_depth: number;
  crotch_length: number;
  waist_to_knee: number;
  waist_to_floor: number;
  side_waist_to_floor: number;
}

export type MeasurementField = keyof FullMeasurements;

File: old/backend/app/schemas/measurements.py
from pydantic import BaseModel


class MeasurementsResponse(BaseModel):
    """Full body measurements response."""
    back_waist_length: float
    front_waist_length: float
    full_bust: float
    bust_height: float
    half_bust_point_distance: float
    full_waist: float
    small_hip: float
    full_hip: float
    neck_circumference: float
    half_back_width: float
    half_front_width: float
    shoulder_length: float
    armhole_circumference: float
    underarm_height: float
    arm_length: float
    upper_arm: float
    elbow_height: float
    wrist: float
    waist_to_hip: float
    crotch_depth: float
    crotch_length: float
    waist_to_knee: float
    waist_to_floor: float
    side_waist_to_floor: float
